Feature/morescenarios

steveblandino opened the pull request.

neerajvarshney committed:
examples added

neerajvarshney committed:
merged

neerajvarshney committed:
Error fixed in tree travesal

For thired order reflections

neerajvarshney committed:
boxLectureRoom Node Positions

Node positions changed

neerajvarshney committed:
all examples file checked

neerajvarshney committed:
flowchart integration

neerajvarshney committed:
Update parameterCfg.m

neerajvarshney committed:
material lib files

neerajvarshney committed:
merged checked again

neerajvarshney committed:
test suite failed

neerajvarshney committed:
test suite failed

steveblandino commented on src/examples/DataCenter/Output/Ns3/NodesPosition/NodesPosition.csv:
```diff
@@ -0,0 +1,2 @@
+8.39,15.4,2.19
```
How come this file is here? I think we do not need it anymore

neerajvarshney replied:
ll generate for all the examples

steveblandino commented on src/examples/DataCenter/Output/Ns3/QdFiles/Tx0Rx1.txt:
```diff
@@ -1,8 +1,151 @@
-267.553,91.6618,330.81,268.641,267.553,269.427,39.4456,91.0768,314.833,268.641,135.194,91.6618,330.81
-90,90,90,90,86.1549,90,90,90,90,87.8608,90,87.3857,88.1208
-87.5534,88.3382,29.1901,271.359,87.5534,89.4271,39.4456,271.077,314.833,271.359,135.194,88.3382,29.1901
+0
```
How come there is a txt output while the input config is set to json?

neerajvarshney replied:
ll generate for all the examples

steveblandino commented on src/examples/DataCenter/Output/Visualizer/Mpc.json:
```diff
@@ -1,3 +1,8 @@
 {"TX":0,"PAA_TX":0,"RX":1,"PAA_RX":0,"Rorder":0,"MPC":[[8.39,15.4,2.19,7.92,4.4,2.19]]}
+<<<<<<< HEAD
```
This conflict was not solved. Rerun this scenario

neerajvarshney replied:
ll generate for all the examples

steveblandino commented on src/main.m:
```diff
@@ -50,8 +55,11 @@
-scenarioNameStr = '';
+% Leave Empty for Default 'ScenarioTest' e.g., scenarioNameStr = '';
+% To Run Existing Scenario Given in 'src/examples/' Folder, 
+% Set scenarioNameStr = 'examples/FolderName' 
```
Leave it as before:  scenarioNameStr = '';

neerajvarshney replied:
Done

steveblandino commented on src/main.m:
```diff
@@ -79,11 +87,14 @@
 % Input System Parameters
 paraCfg = parameterCfg(scenarioNameStr);
-% Input Node related parameters
+
```
Remove empty line

neerajvarshney replied:
Done

steveblandino commented on src/main.m:
```diff
@@ -79,11 +87,14 @@
+% Input Node Related Parameters
 [paraCfg, nodeCfg] = nodeProfileCfg(paraCfg);
-% Run raytracing function and generate outputs
+
```
Remove empty line

neerajvarshney replied:
Done

steveblandino commented on src/main.m:
```diff
@@ -79,11 +87,14 @@
 outputPath = Raytracer(paraCfg, nodeCfg);
 
 fprintf('Save output data to:\n%s\n',outputPath);
+toc(t0);
```
Remove

neerajvarshney replied:
Done but I removed the one which is after SIMULATION COMPLETE final display

steveblandino commented on src/main.m:
```diff
@@ -1,46 +1,51 @@
 clear
 close all
 clc
+warning('on') % to enable all warnings
```
The default warning state is 'on'. No need of this

neerajvarshney replied:
Leave it because I was not getting warning.

neerajvarshney replied:
There is no harm in adding warning on

steveblandino replied:
I think is better to avoid adding code that is not really needed. You can use it localy rather than pushing it into the repo

neerajvarshney replied:
I removed but I will check on bd14 where I was not getting warning. Thats why I included this.

steveblandino commented on src/parameterCfg.m:
```diff
@@ -72,17 +68,39 @@
 % Switch to turn ON or OFF the Qausi dterministic module
 % 1 = ON, 0 = OFF (Default)
-para = fieldToNum(para, 'switchQDGenerator', [0,1], 0);
+para = fieldToNum(para, 'switchDiffuseComponent', [], 0);
```
Why option validValues= [0,1] has been removed?

neerajvarshney replied:
added

steveblandino commented on src/parameterCfg.m:
```diff
@@ -72,17 +68,39 @@
+% Switch to consider only diffuse components up to 
+% diffusePathGainThreshold (in dB) below the deterministic ray. This switch 
+% is only used for NIST measurement based scenarios. Default value is set 
+% as inf which means software does not discard any diffuse components
```
inf or -inf?

neerajvarshney replied:
updated to -inf

steveblandino commented on src/parameterCfg.m:
```diff
@@ -72,17 +68,39 @@
 para = fieldToNum(para, 'totalNumberOfReflections', [], 2);
+if strcmp(para.switchQDModel,'tgayMeasurements') ...
+        && para.totalNumberOfReflections>2
+    warning(strcat('totalNumberOfReflections for switchQDModel = tgayMeasurements',...
```
Why? Is this documented?

neerajvarshney replied:
angle of incident is computed till second order of reflection. I will include this in document

steveblandino commented on src/parameterCfg.m:
```diff
@@ -107,8 +125,17 @@
-if ~isfield(para, 'materialLibraryPath')
-    warning('Environment file path not defined. Using default material library.')
+if isfield(para, 'materialLibraryPath') 
+currentPath =pwd;
```
Use matlab automatic indentation (select and press ctrl+I)

neerajvarshney replied:
done

steveblandino commented on src/parameterCfg.m:
```diff
@@ -107,8 +125,17 @@
+materialLibraryAbsPath = fullfile(currentPath(1:find((pwd==filesep)==1, ...
+                            1, 'last')), 'src',para.materialLibraryPath);
+end
+if ~isfield(para, 'materialLibraryPath') || ~isfile(materialLibraryAbsPath)...
```
This shouuld be included inside previous if statement. If materialLibraryAbsPath cannot be reached you will have error in this line

neerajvarshney replied:
I have updated this

steveblandino replied:
I cannot see the update

[Neeraj] I dont know whY I cannot comment on this but here is my answer.
I had corrected please check.

neerajvarshney replied:
I dont know whY I cannot comment on this but here is my answer.
I had corrected please check.

steveblandino commented on src/parameterCfg.m:
```diff
@@ -189,4 +216,23 @@
+
+end
+
+function isMaterialLibraryFileFormat = isMaterialLibraryFileFormat(para,...
```
You can give in input switchQdModel rather than the whole structure

neerajvarshney replied:
Done

steveblandino commented on src/parameterCfg.m:
```diff
@@ -189,4 +216,23 @@
+    materialLibrary = importMaterialLibrary(para.materialLibraryPath);
+    if strcmp(para.switchQDModel,'tgayMeasurements') && ...
+            size(materialLibrary,2) == 3
+        isMaterialLibraryFileFormat = true(1);
```
Just true is fine

neerajvarshney replied:
Done

steveblandino commented on src/parameterCfg.m:
```diff
@@ -189,4 +216,23 @@
+        isMaterialLibraryFileFormat = true(1);
+    elseif strcmp(para.switchQDModel,'nistMeasurements') && ...
+            size(materialLibrary,2) == 26
+        isMaterialLibraryFileFormat = true(1);
```
Just true is fine

neerajvarshney replied:
Done

steveblandino commented on src/parameterCfg.m:
```diff
@@ -189,4 +216,23 @@
+            size(materialLibrary,2) == 26
+        isMaterialLibraryFileFormat = true(1);
+    else
+        isMaterialLibraryFileFormat = false(1);
```
false

neerajvarshney replied:
Done

steveblandino commented on src/parameterCfg.m:
```diff
@@ -189,4 +216,23 @@
+        isMaterialLibraryFileFormat = false(1);
+    end
+else
+    isMaterialLibraryFileFormat = false(1);
```
false

neerajvarshney replied:
Done

steveblandino commented on src/raytracer/LinearMobility.m:
```diff
@@ -23,39 +23,41 @@
-% and should note the date and nature of any such change. Please explicitly
-% acknowledge the National Institute of Standards and Technology as the source of the
-% software.
+%--------------------------Software Disclaimer-----------------------------
```
Is LinearMobility still in use? maybe it could be discarded

neerajvarshney replied:
I did not check it. Let me know if we need to remove

neerajvarshney replied:
I sent email to Mattia, let us wait.

steveblandino commented on src/raytracer/Raytracer.m:
```diff
@@ -185,16 +186,20 @@
                                 nMpc > 0                            
-                            multipath1 = multipathTemporary(:,...
-                                2:end); %Discard reflection order column
+                                multipath1 = multipathTemporary(:,...
```
Remove white space

neerajvarshney replied:
done

steveblandino commented on src/raytracer/getTgayReflectionLoss.m:
```diff
@@ -0,0 +1,108 @@
+function reflectionLossdB = getTgayReflectionLoss(MaterialLibrary,...
+                materialIndex, multipath)
+% getReflectionLossApproach2 returns reflection loss in dB. 
```
Wrong function name

neerajvarshney replied:
done

steveblandino commented on src/raytracer/getTgayReflectionLoss.m:
```diff
@@ -0,0 +1,108 @@
+function reflectionLossdB = getTgayReflectionLoss(MaterialLibrary,...
+                materialIndex, multipath)
+% getReflectionLossApproach2 returns reflection loss in dB. 
+% This function is based on the MATLAB WLAN toolbox.
```
Remove this

steveblandino commented on src/raytracer/getTgayReflectionLoss.m:
```diff
@@ -0,0 +1,108 @@
+reflectionCoefficient = getReflectance(MaterialLibrary,materialIndex,multipath);
+% Calculate reflection loss. It is the sum (in dB) of the mean
+% of the V and H reflectances for each reflection.
+reflectionLoss = prod(sum(reflectionCoefficient)/2);
```
sum/2 = mean

neerajvarshney replied:
done

steveblandino commented on src/raytracer/getTgayReflectionLoss.m:
```diff
@@ -0,0 +1,108 @@
+% Calculate reflection loss. It is the sum (in dB) of the mean
+% of the V and H reflectances for each reflection.
+reflectionLoss = prod(sum(reflectionCoefficient)/2);
+reflectionLossdB = -20*log10(abs((reflectionLoss))); 
```
Remove parentheses

neerajvarshney replied:
done

steveblandino commented on src/raytracer/getTgayReflectionLoss.m:
```diff
@@ -0,0 +1,108 @@
+    elseif  reflectionOrderIndex == 2
+        reflectionMaterialIndex = materialIndex(1,2);
+    else
+        error(strcat('Incident angles are obtained till second order ',...
```
Is this true? Don't we describe MPCs above the second order?

neerajvarshney replied:
We cannot consider Tgay since we are obtaining angle of incident till second order reflection. However it will not go in else loop since we are fixing 2 in parametercnf.m if reflectio order is set >2 for tgay channel

steveblandino commented on src/raytracer/multipath.m:
```diff
@@ -83,7 +91,7 @@
 nVarOut = 21;
 sizeArrayOfPlanes = size(ArrayOfPlanes);
-dRay = zeros(1, nVarOut);
+% dRay = zeros(1, nVarOut);
```
why is this commented?

neerajvarshney replied:
This variable should be zero when we are considering new iterateNumberOfRowsArraysOfPlanes. I placed it in correct location

steveblandino commented on src/raytracer/qdGenerator.m:
```diff
@@ -33,33 +72,42 @@
 % Pre/post cursors output
-outputPre = getQdOutput(cursorOutput, arrayOfMaterials, materialLibrary, 'pre');
-outputPost = getQdOutput(cursorOutput, arrayOfMaterials, materialLibrary, 'post');
+switch qdModelSwitch 
```
Use auto-indent

neerajvarshney replied:
done

steveblandino commented on src/raytracer/qdGenerator.m:
```diff
@@ -33,33 +72,42 @@
 
-
 %% Utils
+
```
Remove white line

neerajvarshney replied:
done

steveblandino commented on src/raytracer/qdGenerator.m:
```diff
@@ -181,4 +229,79 @@
+    taus = nan(intraClusterParams.n+1, 1);
+    taus(1) = dRayOutput(8); 
+    i = 1;
+    while i <= intraClusterParams.n
```
You can use for loop here

neerajvarshney replied:
I dont want to change anything to avoid errors at this stage.

neerajvarshney replied:
We dont have time for testing again.

steveblandino replied:
There no need to test for vs while, it's just easier to read. Remove line 246 and 251 and while with for i = 1: intraClusterParams.n. Which is exactly what you did below

steveblandino commented on src/raytracer/qdGenerator.m:
```diff
@@ -181,4 +229,79 @@
+    end
+    
+    % generate path gain, doa/dod, AOA/AOD (AZ & EL) and phase offset 
+    for i = 1:intraClusterParams.n
```
Can  this loop be merged with the previous one?

neerajvarshney replied:
No

steveblandino replied:
Why not, I think you are doing the same loop. In the first loop filling output(i,8) in the second filling the other columns.

steveblandino commented on src/raytracer/qdGenerator.m:
```diff
@@ -181,4 +229,79 @@
+    % generate path gain, doa/dod, AOA/AOD (AZ & EL) and phase offset 
+    for i = 1:intraClusterParams.n
+        output(i, 1) = (dRayOutput(1)+(i)./10^ceil(log10(intraClusterParams.n))); 
+        output(i, 9) =  pow2db((db2pow(dRayOutput(9))...
```
db2pow(a)/db2pow(b) = db2pow(a-b)

neerajvarshney replied:
done

steveblandino commented on src/raytracer/treetraversal.m:
```diff
@@ -139,22 +134,14 @@
             % and points in Array_of_planes and Array_of_points
             plane2 = ArrayOfPlanes(numberOfPlanes,indexPlanesTemporary - 4:indexPlanesTemporary - 1);
-            normal2(1) = plane2(1:3);
+%             normal2(1) = plane2(1:3);
```
Delete if not needed

neerajvarshney replied:
deleted, but why in the next line we still have (1)? I corrected this error in this version. May be I did something wrong in merge.
For your reference, it should be             normal2 = plane2(1:3); otherwise it would give you error for third order of reflection. We had corrected this error in my previous PR but somehow this error propagates till now.

steveblandino commented on src/utils/readNodePositions.m:
```diff
@@ -0,0 +1,32 @@
+function out = readNodePositions(path)
```
Is this function needed?

neerajvarshney replied:
removed

steveblandino commented on test/testSuite.m:
```diff
@@ -22,12 +22,12 @@
-%     'OutputDetail', 3);
-
-% Runs all procedures from exampleOutputsTest
+% Runs a single precedure from exampleOutputsTest
```
Keep old file

neerajvarshney replied:
I will do it once I am done with test suite. We just need to uncomment and comment lines

steveblandino reviewed (commented):
Hi @neerajvarshney, I have added some comments in your code. Please revise your code, I'll check it again before submitting it to Mattia.
Regarding the sceanarios, I think the output should be re-generated, I see a mixture of previous and new output format

neerajvarshney committed:
review comments addressed

neerajvarshney committed:
Update parameterCfg.m

steveblandino commented on src/main.m:
```diff
@@ -51,7 +56,7 @@
 %% Input
 % Leave empty for default 'ScenarioTest'
-scenarioNameStr = '';
+scenarioNameStr = 'examples/LivingRoom';
```
Leave empty string

neerajvarshney replied:
Will correct when I do final commit

steveblandino commented on src/raytracer/multipath.m:
```diff
@@ -131,32 +130,49 @@
             end
         end
-
+%         if isMpc == 1
```
Remove commented part if not needed

neerajvarshney replied:
DONE

steveblandino commented on src/raytracer/qdGenerator.m:
```diff
@@ -255,8 +254,8 @@
         output(i, 1) = (dRayOutput(1)+(i)./10^ceil(log10(intraClusterParams.n))); 
-        output(i, 9) =  pow2db((db2pow(dRayOutput(9))...
-                        /db2pow(intraClusterParams.Kfactor)).*...
+        output(i, 9) =  pow2db((db2pow(dRayOutput(9)...
```
Remove extra parentheses after pow2db

neerajvarshney replied:
there is no extra (

steveblandino reviewed (commented):
@neerajvarshney , good work, only few other comments from my side

neerajvarshney committed:
city block example

neerajvarshney committed:
test suite failed

neerajvarshney committed:
test suite passed

neerajvarshney committed:
testSuite.m

neerajvarshney committed:
cityblock JSON output

neerajvarshney committed:
Code cleaning

neerajvarshney committed:
Document, Code cleaning, re-run scenarios

neerajvarshney committed:
Padova Comments

neerajvarshney committed:
Box.xml renamed to BoxLectureRoom.xml

1. Box.xml renamed to BoxLectureRoom.xml for consistency
2. Removed Rotation [0 0 0] files from each examples (useless)
3. updated document for boxLectureRoom.xml

neerajvarshney committed:
Code cleaning

neerajvarshney committed:
removed BoxLectureRoom and SimplifiedParkingLot

neerajvarshney committed:
added BoxLectureRoom and SimplifiedParkingLot

steveblandino commented:
Review completed
